docs/adr/0014-on-chain-state-management.md

```diff
@@ -0,0 +1,162 @@
+
+A transaction qualifies as a `CollectCom` if and only if:
+
+- There's exactly one input spending funds locked by a script `νHead` in an `Initial` state, using a `CollectCom` redeemer. 
+
+From observing a `CollectCom` transaction, observers may infer the following pieces of information:
+
+- A new UTXO entry locked by a `νHead` script;
+- Some now-spent-UTXO entries which were locked by `νCommit` scripts;
+- An initial UTXO set (and a corresponding root hash for that UTXO set).
+
+#### Observing a 'Close' transaction
+
+A transaction qualifies as a `Close` if and only if:
+
+- There's exactly one input spending funds locked by a script `νHead` in an `Open` state, using a `Close` redeemer. 
+
+From observing a `Commit` transaction, observers may infer the following pieces of information:
+
+- A new UTXO entry locked by a `νHead` script;
+- A candidate final UTXO set
+- A candidate snapshot number 
+
+#### Observing a 'Fanout' transaction
+
+A transaction qualifies as a `Fanout` if and only if:
+
+- There's exactly one input spending funds locked by a script `νHead` in a `Closed` state, using a `Fanout` redeemer. 
+
+From observing a `Commit` transaction, observers may infer the following pieces of information:
+
+- A candidate final UTXO set
+- A candidate snapshot number 
+
+### Observing Transitions
+
+One interesting detail to notice is how any transition can be observed _by itself_, out of any context. This is actually consequence of Cardano's smart-contract design which forces transactions validations to be deterministic and solely dependent on the surrounding transaction. This means that it is possible to write a pure function `observeTransition` for observing transitions as such:
+
+```hs
+observeTransition ::
+  Configuration ->
+  Tx ->
+  Maybe Transition
+
+data Configuration = Configuration
+  { knownScripts :: KnownScripts
+  , ourCardanoKey :: SigningKey PaymentKey 
+  , ourEphemeralKey :: SigningKey EphemeralKey
+  }
+
+data KnownScripts = KnownScripts
+  { νHead :: Script 
+  , νInitial :: Script 
+  , νCommit :: Script 
+  }
+
+data Transition 
+  = Init 
+      { headParameters :: HeadParameters
+      , ephemeralKeyHashes :: [VerificationKeyHash]
+      , initials :: Map TxOutRef (TxOut CtxUTxO)
+      }
+  | Commit
+      { initialConsumed :: TxOutRef
+      , commit :: (TxOutRef, TxOut CtxUTxO)
+      }
+  | TODO...
+```
+
+In particular, we can also imagine splitting that function into sub-functions each dedicated to each transition. A corollary of this means that it is possible to represent our current state as a list of transaction `[Tx]` from which we can extract on-demand the required information for the application logic by folding over the sequence of transactions with the `observeTransition` function.
+
+More exactly, in the light of upcoming work stream, we would want to also keep track of the slots at which transactions were found in blocks (i.e. `[(SlotNo, Tx)]`) as this allows for easy rollback later on. 
```

Comment:
This would translate to one `SlotNo` and `Tx` per chain event, right?

Reply:
No, hence the list :smile:. I think we will probably not hit a case where we would have more than one per slot, but it's also plausible.